pydmd/havok.py

```diff
@@ -0,0 +1,194 @@
+"""
+Derived module from hankeldmd.py for HAVOK.
+
+Reference:
+- S. L. Brunton, B. W. Brunton, J. L. Proctor,Eurika Kaiser, and J. N. Kutz.
+Chaos as an intermittently forced linear system.
+Nature Communications, 8, 2017.
+"""
+
+import numpy as np
+from scipy import signal
+from .hankeldmd import HankelDMD
+from .utils import compute_svd
+
+class HAVOK(HankelDMD):
+    """
+    Hankel alternative view of Koopman (HAVOK) analysis
+    """
+    def __init__(
+        self,
+        svd_rank=0,
+        tlsq_rank=0,
+        exact=False,
+        opt=False,
+        rescale_mode=None,
+        forward_backward=False,
+        sorted_eigs=False,
+        tikhonov_regularization=None,
+        d=1,
+    ):
+        super().__init__(
+            svd_rank=svd_rank,
+            tlsq_rank=tlsq_rank,
+            exact=exact,
+            opt=opt,
+            rescale_mode=rescale_mode,
+            forward_backward=forward_backward,
+            sorted_eigs=sorted_eigs, 
+            tikhonov_regularization=tikhonov_regularization,
+            d=d,
+        )
+        self._embeddings = None
+        self._A = None
+        self._B = None
+
+    @property
+    def linear_embeddings(self):
+        """
+        Get the time-delay embeddings of the data that are governed by linear
+        dynamics. Emeddings are stored as columns of the returned matrix.
+
+        :return: matrix containing the linear time-delay embeddings.
+        :rtype: numpy.ndarray
+        """
+        return self._embeddings[:,:-1]
```

Comment:
This will trigger a weird error if `fit()` is not called

```suggestion
        if self._embeddings is None:
                raise RuntimeError('fit() not called')
        return self._embeddings[:,:-1]
```